### DISCUSSION TITLE
How to add images in interactive mode?

### DISCUSSION THREAD

**Discussion by notooth1** [ggml-org/llama.cpp#7690] [Q&A]
I want to add images in an interactive mode with a multimodal model. How can I do that?

**Comment by zeitsmatt** [comment]
I'm interested in this as well.  I found the spot in the code where the image and system prompt are processed.  The trick is moving all of the corresponding code from main.cpp over to llava-cli.cpp.  I don't know how much marshalling is involved in handling all of the parameters.  I'm thinking of just getting a simple repl style loop going and then after showing how useful it is, maybe the effort will be put in to move the existing code over or move the llava code into the main codebase.
